crates/jiter/src/number_decoder.rs

```diff
@@ -398,25 +398,49 @@ impl AbstractNumberDecoder for NumberRange {
         };
 
         index += 1;
-        while let Some(next) = data.get(index) {
-            match next {
-                b'0'..=b'9' => (),
-                b'.' => {
+        for _ in 0..18 {
+            if let Some(digit) = data.get(index) {
+                if INT_CHAR_MAP[*digit as usize] {
+                    index += 1;
+                    continue;
+                } else if matches!(digit, b'.') {
                     index += 1;
                     let end = consume_decimal(data, index)?;
                     return Ok((start..end, end));
-                }
-                b'e' | b'E' => {
+                } else if matches!(digit, b'e' | b'E') {
                     index += 1;
                     let end = consume_exponential(data, index)?;
                     return Ok((start..end, end));
                 }
-                _ => break,
             }
-            index += 1;
+            return Ok((start..index, index));
+        }
+        loop {
+            let (chunk, new_index) = IntChunk::parse_big(data, index);
```

Comment:
As I understand it, this is just adding the SIMD code to range to allow skipping to be faster?

Reply:
yes.